### PR TITLE
Redshift scraper

### PR DESCRIPTION
**Pull request by jariavikainen**
Add scripts and functionality needed for scraping metadata from Redshift to the data catalog. The change in the `databuilder/databuilder/extractor/neptune_search_data_extractor.py` is to fix the way Neptune contents are indexed to Elasticsearch. The keys used in indexing were incorrect, which caused errors in the dashboard

### CODE REVIEW

**Comment by evakkuri** [comment]
Please add a description of main features in this PR

File: databuilder/README YLE.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Amundsen databuilder
+
+This library is used for ingesting data into Yle data catalog.
```
Comment (evakkuri) [inline databuilder/README YLE.md:3]: Would be great to add links here to the data catalog itself, and / or other relevant repos

File: databuilder/README YLE.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+## Setup
+
+1. Set up a dedicated virtual environment for python, e.g. using [pyenv-virtualenv](https://github.com/pyenv/pyenv-virtualenv).
+2. Build and install amundsen databuilder from sources
+`pip install '.[neptune, redshift]'`
+
+## Configuration
+
+Modify the configuration in `databuilder/config/databuilder.yml`.
```
Comment (evakkuri) [inline databuilder/README YLE.md:13]: How are these used when running in a server?

File: databuilder/tests/unit/transformer/test_table_hashtag_transformer.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import unittest
+
+from databuilder.models.table_metadata import TableMetadata
+from databuilder.transformer.table_hashtag_transformer import TableHashtagTransformer
+
+
+class TestTableHashtagTransformer(unittest.TestCase):
+    def test_single_tag(self) -> None:
```
Comment (evakkuri) [inline databuilder/tests/unit/transformer/test_table_hashtag_transformer.py:8]: Please add short desriptions what these tests are testing

File: databuilder/tests/unit/transformer/test_table_owner_transformer.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+from pyhocon import ConfigFactory
+
+from databuilder.models.table_metadata import TableMetadata
+from databuilder.models.table_owner import TableOwner
+from databuilder.transformer.table_owner_transformer import TableOwnerTransformer
+
+
+class TestTableOwnerTransformer(unittest.TestCase):
+    def test_table_is_not_modified(self) -> None:
```
Comment (evakkuri) [inline databuilder/tests/unit/transformer/test_table_owner_transformer.py:12]: Please add short descriptions what these tests are testing

File: databuilder/scripts/scrape_redshift.py [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+
+configuration = yaml.safe_load(open('./config/config.yml', 'r'))
+es_host = configuration['ElasticsearchHost']
+es = Elasticsearch([
+    {'host': es_host},
+])
+
+
+def create_redshift_job(current_publish_tag):
+    tmp_folder = '/var/tmp/amundsen/table_metadata'
```
Comment (evakkuri) [inline databuilder/scripts/scrape_redshift.py:35]: Please add docstring for this function

File: databuilder/scripts/scrape_redshift.py [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+    })
+    job = DefaultJob(
+        conf=job_config,
+        task=task,
+        publisher=publisher)
+    return job
+
+
+def create_stale_data_removal_job(target_nodes=[], target_relations=[], dry_run=False):
+    """
```
Comment (evakkuri) [inline databuilder/scripts/scrape_redshift.py:90]: Please add short description what this function does, in addition to the parameters (which are great already)

File: databuilder/scripts/scrape_redshift.py [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+        task=task)
+    return job
+
+
+def create_es_publisher_job(elasticsearch_index_alias='table_search_index',
+                            elasticsearch_doc_type_key='table',
+                            model_name='databuilder.models.table_elasticsearch_document.TableESDocument',
+                            entity_type='table',
+                            elasticsearch_mapping=None):
+    """
```
Comment (evakkuri) [inline databuilder/scripts/scrape_redshift.py:122]: Please add short description what this function does, in addition to the parameters (which are great already)

File: databuilder/scripts/scrape_redshift.py [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+    job = DefaultJob(
+        conf=job_config,
+        task=task,
+        publisher=ElasticsearchPublisher()
+    )
+    return job
+
+
+def main():
+    # Uncomment next line to get INFO level logging
```
Comment (evakkuri) [inline databuilder/scripts/scrape_redshift.py:186]: "Uncomment next line" -> should it be commented?

File: databuilder/scripts/index_to_es.py [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+    {'host': es_host},
+])
+
+
+def create_es_publisher_job(elasticsearch_index_alias='table_search_index',
+                            elasticsearch_doc_type_key='table',
+                            model_name='databuilder.models.table_elasticsearch_document.TableESDocument',
+                            entity_type='table',
+                            elasticsearch_mapping=None):
+    """
```
Comment (evakkuri) [inline databuilder/scripts/index_to_es.py:30]: Please add short description on what this function does in addition to the parameters

File: databuilder/scripts/index_to_es.py [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+    job = DefaultJob(
+        conf=job_config,
+        task=task,
+        publisher=ElasticsearchPublisher()
+    )
+    return job
+
+
+def main():
+    # Uncomment next line to get INFO level logging
```
Comment (evakkuri) [inline databuilder/scripts/index_to_es.py:94]: "Uncomment next line" -> should it be commented?

File: databuilder/config/config.yml [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+
+AwsRegionName: 'eu-west-1'
+NeptuneS3BucketName: 'example-neptune-bucket-name'
+NeptuneS3BucketPath: 'example-neptune-s3-path'
+NeptuneRoleName: 'example-neptune-role'
+
+
+# Redshift configuration
+RedshiftUser: 'username'
+RedshiftPassword: 'password'
```
Comment (evakkuri) [inline databuilder/config/config.yml:18]: How is this used when running in a server?

File: databuilder/README YLE.md [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+
+## Run tests
+1. Set up a dedicated virtual environment for python, e.g. using [pyenv-virtualenv](https://github.com/pyenv/pyenv-virtualenv).
+2. `pip install --upgrade pip wheel`
+3. Install libraries needed for running the tests (in databuilder directory)  
+`pip install ../common && pip install '.[test]'`  
+Pip may complain about incorrect certifi version, but that shouldn't be a problem. If you want, you can downgrade certifi manually:  
+`pip install 'certifi<2021.0.0'`
+4. Run tests:  
+`make test`
```
Comment (evakkuri) [inline databuilder/README YLE.md:31]: Is this something deployed to server? If so, how to do deployment?

**Review by evakkuri — COMMENTED** [review]
Added some comments. Good that there are some comments and docstrings already, but I would like to see some short descriptions on top of that. Naturally it's difficult to make give an educated review when I don't know this product, but on the other hand it's vital that everyone gets to know this, if this is something that our team will maintain -> I think descriptions help :)

File: databuilder/tests/unit/transformer/test_table_hashtag_transformer.py (hunk 3 above)
Comment (jariavikainen) [reply databuilder/tests/unit/transformer/test_table_hashtag_transformer.py]: Changed some of the test names into more descriptive ones. I feel that together with the fairly simple testing code that should be enough

File: databuilder/README YLE.md (hunk 2 above)
Comment (jariavikainen) [reply databuilder/README YLE.md:17]: I updated the description above a bit. Do you think it is enough to have the link to the packer config (which does configure the databuilder to be included in the AMI)?

**Review by evakkuri — APPROVED** [review]
Very good additions 👍 I would still add some info on when the scraper should be run from local, so that it's clearer when to use this. I still don't know what the tests are supposed to test, but that's also due to me not being familiar with the product, so that's fine. Approved anyways.

**Pull request merged**